src/components/share-drawer/ShareDrawer.tsx

```diff
@@ -89,55 +79,76 @@ export const ShareDrawer = () => {
-  const track = useSelectorWeb(getTrack)
-  const artist = useSelectorWeb((state: CommonState) =>
-    getUser(state, { id: track?.owner_id })
-  )
+  const { content, source } = useSelectorWeb(getShareState)
   const account = useSelectorWeb(getAccountUser)
-  const source = useSelectorWeb(getSource)
 
   const { toast } = useContext(ToastContext)
 
   const isOwner = Boolean(
-    account && artist && account.user_id === artist.user_id
+    content?.type === 'track' &&
+      account &&
+      account.user_id === content.artist.user_id
   )
+  const shareType = content?.type ?? 'track'
 
   const handleClose = useCallback(() => {
     dispatchWeb(setVisibility({ modal: 'Share', visible: false }))
   }, [dispatchWeb])
 
   const handleShareToTwitter = useCallback(async () => {
-    const twitterText = `Check out ${track.title} by ${artist.handle} on @AudiusProject #Audius`
-    const trackUrl = getTrackRoute(track, true)
-    const twitterShareUrl = getTwitterLink(trackUrl, twitterText)
+    if (!content) return
+    const twitterShareUrl = getTwitterShareUrl(content)
     const isSupported = await Linking.canOpenURL(twitterShareUrl)
     if (isSupported) {
       Linking.openURL(twitterShareUrl)
     } else {
       console.error(`Can't open: ${twitterShareUrl}`)
     }
+
     handleClose()
-  }, [track, artist, handleClose])
+  }, [content, handleClose])
 
   const handleShareToTikTok = useCallback(() => {
-    dispatchWeb(requestOpenTikTokModal({ id: track.track_id }))
+    if (content?.type === 'track') {
+      dispatchWeb(requestOpenTikTokModal({ id: content.track.track_id }))
+    }
     handleClose()
-  }, [dispatchWeb, track, handleClose])
+  }, [content, dispatchWeb, handleClose])
 
   const handleCopyLink = useCallback(() => {
-    const trackUrl = getTrackRoute(track, true)
-    Clipboard.setString(trackUrl)
+    if (!content) return
+    const link = getContentUrl(content)
+    Clipboard.setString(link)
     toast({ content: 'Copied Link to Track', type: 'info' })
     handleClose()
-  }, [toast, track, handleClose])
+  }, [toast, content, handleClose])
 
   const handleOpenShareSheet = useCallback(() => {
-    dispatchWeb(shareTrack(track.track_id, source))
+    if (!source || !content) return
+    switch (content.type) {
+      case 'track':
+        dispatchWeb(shareTrack(content.track.track_id, source))
+        break
+      case 'profile':
+        dispatchWeb(shareUser(content.profile.user_id, source))
+        break
+      case 'album':
+        dispatchWeb(shareCollection(content.album.playlist_id, source))
+        break
+      case 'playlist':
+        dispatchWeb(shareCollection(content.playlist.playlist_id, source))
+        break
+    }
     handleClose()
-  }, [dispatchWeb, track, source, handleClose])
-
-  const shouldIncludeTikTokAction =
-    isOwner && !track.is_invalid && !track.is_delete
+  }, [dispatchWeb, content, source, handleClose])
+
+  const shouldIncludeTikTokAction = Boolean(
```

Comment:
What does `Boolean()` do here - isn't it already a boolean?

Reply:
yeah no reason here! in audius-client i was seeing it was like `boolean | null` which was messing up the prop type of just `boolean`